0010-global-init.md

```diff
@@ -0,0 +1,110 @@
+# Global SDK initialization
+
+*Status: proposed*
+
+Specify the behavior of OpenTelemetry APIs and implementations at startup.
+
+## Motivation
+
+OpenTelemetry is designed with a separation between the API and the
+SDK which implements it, allowing an application to configure and bind
+any compatible SDK at runtime.  OpenTelemetry is designed to support
+"zero touch" instrumentation for third party libraries through the use
+of a global instance.
+
+In many programming environments, it is possible for libraries of code
+to auto-initialize, allowing them to begin operation concurrently with
+the main program, e.g., while initializing static program state.  This
+presents a set of opposing requirements: (1) the API supports a
+configurable SDK; (2) third party libraries may use OpenTelemetry
+without configuration.
+
+## Explanation
+
+There are several acceptable ways to address this situation.  The
+feasibility of each approach varies by language.  The implementation
+must select one of the following strategies:
+
+### Service provider mechanism
+
+Where the language provides a commonly accepted way to inject SDK
+components, it should be preferred.  The Java SPI supports loading and
+configuring the global SDK before it is first used, and because of
+this property the service provider mechanism case leaves little else
+to specify.
+
+### Explicit initializer
+
+When it is not possible to ensure the SDK is installed and configured
+before the API is first used, loading the SDK is handed off to the
+user "at the right time", as stated in [Ruby issue
+19](https://github.com/open-telemetry/opentelemetry-ruby/issues/19).
+In this case, a number of requirements must be specified, as discussed
+next.
+
+## Requirements: Explicit initializer
+
+OpenTelemetry specifies that the default implementation is
+non-operational (i.e., a "no-op"), requiring that API method calls
+result in effectively zero instrumentation overhead.  We expect third
+party libraries to use the global SDK before it is installed, which is
+addressed in a requirement stated below.
+
+The explicit initializer method should take independent `Tracer` and
+`Meter` objects (e.g., `opentelemetry.Init(Tracer, Meter)`).  The SDK
```

Comment:
I suggest that we have separate initialization for Tracer and Meter, even in the examples here.

Reply:
I believe it's possible with this API to independently construct and configure the Tracer and Meter that will be used, all that this requires is that they be provided in a single initializer call. I believe that we should encourage a single initializer call so that when a combined implementation (i.e., an SDK that presents both APIs) is registered, it can be sure that its registration is all-or-none.

In any case, it should be possible to provide Tracer and Meter implementations that are completely independent, or to pass a Noop implementation of either in the initializer.